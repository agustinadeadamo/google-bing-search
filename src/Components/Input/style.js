/**
 * @desc Dependencies
 */
import styled from 'styled-components';

/**
 * @desc Variables
 */
import { Colors } from 'Mainstyles/Variables.js';

export const InputComponent = styled.input`
    width: 100%;
    border: 1px solid ${Colors.grey};
    color: ${Colors.grey};
    height: 40px;
    padding: 0 10px;
    margin: 10px 0;
`;
