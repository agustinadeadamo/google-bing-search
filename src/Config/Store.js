/**
 * @desc Dependencies
 */
import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';

/**
 * @desc Middlewares
 */
import Middlewares from './Middlewares';

/**
 * @desc Reducers
 */
import Reducers from 'Reducer/Reducer';

/**
 * @desc Store
 */
const Store = createStore(Reducers, composeWithDevTools(applyMiddleware(...Middlewares)));

export default Store;
