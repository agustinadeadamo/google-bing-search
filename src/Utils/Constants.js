/* Search mode */
export const SearchMode = {
    "GOOGLE" : 'Google',
    "BING" : 'Bing',
    "GOOGLE_BING" : 'Goolge & Bing',
};

export const MessageError = {
    'SERVER_ERROR': 'There was an error in the server',
    'APP_ERROR': 'There was an error in the app',
};

export const Apikeys = {
    'BING_KEY' : '1ef3364ab9ce48b1b180b109e6aab6fb',
    'GOOGLE_KEY': 'AIzaSyB5Mqgm1gE_sAvmQToGkD2fGnOw-Pl1m5w',
    'GOOGLE_CX': '18ca2e163998cf02d'
}

export const Endpoints = {
    'GOOGLE_SEARCH' : 'https://www.googleapis.com/customsearch/v1',
    'BING_SEARCH': 'https://api.bing.microsoft.com/v7.0/search'
}

