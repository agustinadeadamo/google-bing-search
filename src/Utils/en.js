export const English = {
    'ADD_VALUE_TO_SEARCH' : 'Add a value to search',
    'SELECT_OPTION' : 'Select an option',
    'SUBMIT' : 'Submit',
    'GOOGLE_BING_SEARCH' : 'Google & Bing Search',
}