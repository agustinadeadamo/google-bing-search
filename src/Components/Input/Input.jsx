/**
 * @desc Dependencies
 */
import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from "react-redux";

/**
 * @desc Styles
 */
import { InputComponent } from './style';

/**
 * @desc Actions
 */
import { changeShowError } from 'Actions/ErrorActions';

/**
 * @desc Utils
 */
import { MessageError } from 'Utils/Constants';

const Input = (props) => {

  // Destructuring props
  const {
    value,
    maxLength,
    placeholder,
    type,
    onChangeValue,
    label
  } = props;

  // Store
  const dispatch =  useDispatch()

  /**
   * Function that is executed every time value in input changes
   * 
   * @param {EventTarget} event
   */
  const handleChange = (event) => {

        try {
            const valueInput = event.target.value;
            onChangeValue(valueInput);
        } catch(error) {
            dispatch(changeShowError(MessageError.APP_ERROR));
        }
  };

  return (
        <>
            
            <label>
                {label}
            </label>

            <InputComponent
                onChange={(event) => handleChange(event)}
                value={value}
                maxLength={maxLength}
                type={type}
                placeholder={placeholder}
                data-testid="input-component"
            />
        </>
  );
};

Input.propTypes = {
  value: PropTypes.string.isRequired,
  maxLength: PropTypes.number,
  placeholder: PropTypes.string,
  type: PropTypes.string,
  onChangeValue: PropTypes.func.isRequired,
  label: PropTypes.string
};

Input.defaultProps = {
  maxLength: 100,
  placeholder: '',
  type: 'text',
  value: '',
  onChangeValue: () => {}
};

export default Input;
