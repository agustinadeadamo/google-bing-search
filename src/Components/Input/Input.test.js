/**
 * @desc Dependencies
 */
import { create } from 'react-test-renderer';
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux'

/**
 * @desc Config
 */
import Store from 'Config/Store';

/**
 * @desc Utils
 */
import { English } from 'Utils/en';

/**
 * @desc Components
 */
import Input from './Input';

describe('[Input Component]',() => {

    it('Renders correctly', () => {

        const {queryByTestId} = render(
            <Provider store={Store}>
                <Input onChangeValue={() => {}} value=""/>
            </Provider>
        );

        expect(queryByTestId('input-component')).toBeTruthy();
    
    });

    it('Matches snapshot', () => {

        const input = create(
            <Provider store={Store}>
                <Input onChangeValue={() => {}} value="" />
            </Provider>
        ).toJSON;

        expect(input).toMatchSnapshot();
    
    });

    it('Updates value on change', () => {

        // Mocks function
        const onChangeValue = jest.fn();

        const {queryByTestId} = render(<Provider store={Store}><Input onChangeValue={onChangeValue} value="" /></Provider>);
        
        // Gets input
        const inputComponent = queryByTestId('input-component');

        // Generates change event
        fireEvent.change(inputComponent, {target: {value: 'text'}});
        
        expect(onChangeValue).toHaveBeenCalledTimes(1);
        expect(onChangeValue).toHaveBeenCalledWith('text');
    
    });

    it('Shows correct label', () => {  
            
        const {getByText} = render(
            <Provider store={Store}>
                <Input label={English.ADD_VALUE_TO_SEARCH} onChangeValue={() => {}} value="" />
            </Provider>
        );

        expect(getByText(English.ADD_VALUE_TO_SEARCH)).toBeInTheDocument();

    });

});