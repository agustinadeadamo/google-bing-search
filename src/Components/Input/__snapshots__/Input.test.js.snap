// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Input Component] Matches snapshot 1`] = `[Function]`;
